### PR TITLE
Android/Browser - IME improvements

### LINKED ISSUE #9405: IME editing is still janky on Android
**Describe the bug**
Clicking the auto-completion text box is supposed to enter new words. Instead in Avalonia, we are stuck editing the last word. Moreover, deletion also is not as smooth as expected compared to native controls when deleting a word entered by auto completion.

**To Reproduce**
Use a soft keyboard with IME to edit on Android.

**Expected behavior**
Editing is as smooth as the native control.

**Screenshots**
https://user-images.githubusercontent.com/19922066/201042000-bce99e5c-4b31-4c71-88ae-01ee3cc9a79a.mp4

**Desktop (please complete the following information):**

- OS: Android
- Version 3bfcc3574

### LINKED ISSUE #9428: Android IME - CompositionRegion sync issues
### Discussed in https://github.com/AvaloniaUI/Avalonia/discussions/9423

<div type='discussions-op-text'>

<sup>Originally posted by **lukaszossowski** November 11, 2022</sup>
Hi,

I need to get text input working on mobile, specifically Android. I've used a textbox for this:

`				<TextBox MaxLines="1" TextAlignment="Center" Text={Binding Name} />`

and when I press the textbox it brings up the keyboard. However now there come many issues:

1. Name property has a default value ("Comment") that shows up in the textbox.  When I press any key on the keyboard it copies the default value and then adds the new sign leading to result like this after pressing "b" - "CommentCommentb".
2. When I use the backspace button, it deletes the Commentb part and then places cursor at the beginning of the textbox, but the initial "Comment" still remains. It is ignored until I place the cursor again at its end and then I can delete it.
3. Pressing space deletes existing text and then it enters a weird mode where any new key I press deletes the last one, so my textbox's text now only displays the last letter I pressed and nothing more. The mode works in a way that after pressing a letter I need to press space again to "accept" it and then it moves the cursor one space and now the second letter is displayed as the last key I pressed - until I accept it with spacebar and move on.
4. Pressing OK on the keyboard does nothing, there is no way to actually accept the text that is written.

Overall the experience is mind-boggingly broken and I have no idea what is happening. 

Am I doing something wrong or are those known issues? Has anyone had success with properly working textbox on mobile?</div>

related: #9405

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
This PR improves software keyboard support on android and browser.

## What is the current behavior?
Composition text isn't handled properly on Android. The android input connection is designed to be used with Editable, and text entered in a software keyboard is immediately inserted into the Editable. This allows the software keyboard to switch words under composition without calling `Commit`, which only gets called when Space is pressed to complete a word, or a suggestion is tapped in the software keyboard.
On browser, software keyboards with text prediction or suggestion feature, like the Windows 11 touch keyboard, can remove a word under the cursor, and insert a tapped word suggestion. The insertion is handled currently, but not the deletion, which ends up with the suggestion word being inserted at the cursor.

## What is the updated/expected behavior with this PR?
Composition regions, composing and predictive text, and multiline support now works properly on Android. On browser, word prediction or suggested now replace the selected word.

## How was the solution implemented (if it's not obvious)?
An Editable class is implemented for android, and is passed to the input connection to track edits. `BeginBatchEdit` and `EndBatchEdit` overrides in `AvaloniaInputConnection` are used to signal changes in the editable's text and selection properties.  The text presenter is then updated with the changes.

On browser, `DeleteContentBackwards` is handled, with the selected text being deleted before the word suggestion is inserted.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #9405
Fixes #9428

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029926-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029932-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Browser/Avalonia.Browser/AvaloniaView.cs [hunk 1]
```diff
@@ -342,6 +372,7 @@ private bool OnCompositionEnd(JSObject args)
             if (_client == null)
                 return false;
 
+            Console.WriteLine("composition end");
```
Comment (maxkatz6) [inline src/Browser/Avalonia.Browser/AvaloniaView.cs:375]: Debug leftovers

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029981-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

File: src/Browser/Avalonia.Browser/AvaloniaView.cs (hunk 1 above)
Comment (emmauss) [reply src/Browser/Avalonia.Browser/AvaloniaView.cs]: Resolved.

**Review by Gillibald — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029986-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0029993-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0030019-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**